File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int n = tickets.length;
        int i = 0;
        int time = 0;
        
        while (tickets[k] != 0){
            if (tickets[i] != 0){
                tickets[i]--;
                time++;
            }
            i = (i + 1) % n;
        }
        return time;
    }
}

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        queue = []
        for i, val in enumerate(tickets):
            queue.append([i, val])
        
        time = 0
        while queue:
            index, ticket = queue.pop(0)
            ticket -= 1
            time += 1
            
            if ticket == 0:
                if k == index:
                    return time
            else:
                queue.append([index, ticket])
                
                
        